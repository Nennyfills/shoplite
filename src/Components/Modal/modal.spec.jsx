import React from 'react';
import { shallow } from 'enzyme';

import { checkProps } from 'Test/propsTypes';
import LuminModal from '.';

const props = {
  getTotal: jest.fn(),
  handleIncrease: jest.fn(),
  handleDecrease: jest.fn(),
  handleRemoveItem: jest.fn(),
  products: [],
  cart: [],
  cached: {},
  currencies: 'NGN',
  total: 0,
  visible: false,
  title: '',
};

const setUp = (props = {}) => {
  const component = shallow(<LuminModal {...props} />);
  return component;
};

describe('Lumin Modal', () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });
  it('Should render without errors', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should render Lumin Modal div', () => {
    const wrapper = component.find(`[data-test='lumin-modal']`);
    expect(wrapper.length).toBe(1);
  });
  it('Checking PropTypes should not throw error', () => {
    const propErr = checkProps(LuminModal, props);
    expect(propErr).toBeUndefined();
  });
});
