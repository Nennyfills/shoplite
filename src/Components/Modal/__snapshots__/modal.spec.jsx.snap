// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lumin Modal Should render without errors 1`] = `ShallowWrapper {}`;
